crates/primitives/src/lib.rs

```diff
@@ -56,7 +56,7 @@ pub const SHORT_ADDRESS_CAP: usize = 300;
 #[inline]
 pub fn short_address(address: &Address) -> Option<usize> {
     if address.0[..18].iter().all(|b| *b == 0) {
-        let short_address = u16::from_be_bytes(address.0[18..].try_into().unwrap()) as usize;
+        let short_address = u16::from_be_bytes([address.0[18], address.0[19]]) as usize;
```

Comment:
Noting: an address is 20 bytes, so this is equivalent. 

Hopefully the compiler can figure it out and avoid the bounds checks

Reply:
It is a constant index to an array that is of fixed size, compiler has all the information needed to optimise it